{{cookiecutter.project_slug}}/ios/fastlane/Fastfile

```diff
@@ -139,32 +148,32 @@ platform :ios do
       }
     }
     headers = {
-      "Authorization": "Api-Key #{ENV['WEBHOOK_API_KEY']}",
-      "Content-Type": "application/json"
+      'Authorization': "Api-Key #{ENV['WEBHOOK_API_KEY']}",
+      'Content-Type': 'application/json'
     }
+
     response = HTTParty.post(url, body: data.to_json, headers: headers)
-    puts "API response: #{response.code} #{response.body}"
+    puts("API response: #{response.code} #{response.body}")
 
     response.success?
   end
 end
 
-# get the publicKey of the appetizeApp corresponding to your base_name
 def get_appetize_public_key(platform, base_name)
-  return ENV['APPETIZE_PUBLIC_KEY'] unless ENV['APPETIZE_PUBLIC_KEY'].nil?
+  @appetize_key ||= ENV['APPETIZE_PUBLIC_KEY']
+  @appetize_key ||= fetch_appetize_public_key(platform, base_name)
+  @appetize_key
+end
 
-  response = HTTParty.get("https://#{ENV['APPETIZE_API_TOKEN']}@api.appetize.io/v1/apps")
+def fetch_appetize_public_key(platform, base_name)
+  response =
+    HTTParty.get("https://#{ENV['APPETIZE_API_TOKEN']}@api.appetize.io/v1/apps")
 
-  if response.success?
-    json = JSON.parse(response.body)
+  return nil unless response.success?
 
-    # Find branch name in notes
-    json["data"].each do |value|
-      if value["note"] == base_name && value["platform"] == platform
-        return value["publicKey"]
-      end
-    end
+  item = JSON.parse(response.body)['data'].find do |app_app|
+    app_app['note'] == base_name && app_app['platform'] == platform
   end
 
-  nil
+  item['publicKey']
```

Comment:
Like the other use of `find`, this will throw an exception if it doesn't find an `item`.